### PR TITLE
Store Bearer Token in a tmp file

### PR DESCRIPTION
**Pull request by aleoli**
# Description

This pr fixes a permission problem in writing token files. The pod user is no more root and it has no permission on writing in the pod filesystem.

# How Has This Been Tested?

- [x] locally on KinD

### CODE REVIEW

**Review by palexster — APPROVED** [review]

File: pkg/identityManager/identityManager_test.go [hunk 1]
```diff
@@ -90,8 +85,10 @@ var _ = Describe("IdentityManager", func() {
 		restConfig = cluster.GetCfg()
 
 		namespaceManager = tenantnamespace.NewTenantNamespaceManager(client)
-		identityMan = NewCertificateIdentityManager(cluster.GetClient(), localClusterID, namespaceManager)
-		identityProvider = NewCertificateIdentityProvider(ctx, cluster.GetClient(), localClusterID, namespaceManager)
+		identityMan = NewCertificateIdentityManager(cluster.GetClient(),
+			localClusterID, namespaceManager)
+		identityProvider = NewCertificateIdentityProvider(ctx, cluster.GetClient(),
+			localClusterID, namespaceManager)
```
Comment (giorio94) [inline pkg/identityManager/identityManager_test.go:91]: Are those changes intentional? Would personally avoid too short lines, but that is definitely opinionated. (Same for the occurrences below)

File: pkg/identityManager/tokenManager.go [hunk 2]
```diff
@@ -135,15 +134,28 @@ func (tokMan *iamTokenManager) addClusterID(remoteClusterID string, secret types
...
+	var err error
+	filename, found := tokMan.tokenFiles[remoteClusterID]
+	if found {
+		_, err = os.Stat(filename)
+	}
+
+	if !found || os.IsNotExist(err) {
+		file, err := os.CreateTemp("", "token")
+		if err != nil {
 			klog.Error(err)
```
Comment (giorio94) [inline pkg/identityManager/tokenManager.go:146]: Would enrich a bit the message, to avoid meaningless errors.

**Review by giorio94 — APPROVED** [review]

**Pull request merged**